libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2245,6 +2243,14 @@ bool TypeChecker::visit(IndexAccess const& _access)
 	Expression const* index = _access.indexExpression();
 	switch (baseType->category())
 	{
+	case Type::Category::ArraySlice:
+	{
+		auto const& arrayType = dynamic_cast<ArraySliceType const&>(*baseType).arrayType();
+		if (arrayType.location() != DataLocation::CallData || !arrayType.isDynamicallySized())
+			m_errorReporter.typeError(_access.location(), "Index access is only implemented for slices of dynamic calldata arrays.");
+		baseType = &arrayType;
+		[[fallthrough]];
```

Comment:
Does this not result in an error here: https://github.com/ethereum/solidity/pull/7340/files#diff-9f2a45efaa7b75ab7ec620d3177cf01aR2266 ?

Reply:
Ah sorry, it is a regular index access to an array slice, i.e. `x[:10][2]` - is that right?

Reply:
Yep. By the way so far the start index is non-optional by intention - should it be optional? I don't really mind either way...